client/src/scene/SceneGameArena.ts

```diff
@@ -74,7 +74,9 @@ export class SceneGameArena extends Phaser.Scene {
 
         this.scoreboard = new ScoreboardUI(this, this.socket);
         new ActiveEventsUI(this, this.socket);
-
+        if (this.gameState.playerId != null) {
+            this.trade = new TradeUI(this);
+        }
         if (this.gameState.playerId && this.gameState.playerId >= 0) {
             new ControlsUI(this);
         } else {
```

Comment:
This doesn't need to be in its own if statement, can be just be in the if statement below

Reply:
Actually, that check might be broken. It would return false for the first check if playerId=0.

Can you replace `this.gameState.playerId && this.gameState.playerId >=0` with `this.gameState.playerId != null` and move your trade variable into it

Reply:
Yeah that check is broken. Just changed it, but looks like the Controls UI was still working fine  because it is called when "initPlayer" is received.